### PR TITLE
feat(sdk): add client methods to cli

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Adds the following functionality to the CLI per the V1 -> V2 CLI migration plan:
- run archive
- run unarchive
- run delete
- recurring-run enable
- recurring-run disable
- pipeline get-version
- experiment unarchive

Also:
- cleans up single/double quote inconsistency
- improves informativeness of output for user

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more 
about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by chensun — COMMENTED** [review]
/lgtm

Thanks!

File: sdk/python/kfp/cli/run.py [hunk 1]
```diff
@@ -159,6 +160,60 @@ def get(ctx: click.Context, watch: bool, detail: bool, run_id: str):
     _display_run(client, namespace, run_id, watch, output_format, detail)
 
 
+@run.command()
+@click.argument('run-id')
+@click.pass_context
+def archive(ctx: click.Context, run_id: str):
+    """Archive a run."""
```
[the remark this answers is not in the document]
Comment (connor-mccarthy) [reply sdk/python/kfp/cli/run.py:167]: Thanks and great point. These command docstrings serve a dual purpose: 1) they are standard docstrings and 2) they are the CLI help text.

I reviewed a few popular CLIs and it looks like most use an imperative style:
- `git commit --help` -> "**Create** a new commit containing..."
- `docker build --help` -> "**Build** an image from a Dockerfile"
- `kubectl get pods --help` -> "**Display** one or many resources"
- `gh pr list --help` -> "**List** pull requests..."

So I went with imperative also. I could go either way and happy to make this change, but wanted to document the decision via a comment.

Note: git is actually somewhat inconsistent with this... `git add --help` -> "This command **updates**...".

[the remark this answers is not in the document]
Comment (chensun) [reply sdk/python/kfp/cli/run.py:167]: > 2) they are the CLI help text.

Didn't realize this. What you have is perfectly fine then. :) Thanks for the explanation.

**Pull request merged**